climada_petals/hazard/rf_glofas/test/test_transform_ops.py

```diff
@@ -0,0 +1,662 @@
+        """Test 'return_period' operation"""
+        x = np.arange(10)
+        # Distort the coordinates to test the reindexing
+        x_var = x + self.rng.uniform(low=-1e-7, high=1e-7, size=x.shape)
+        x_var_big = x + self.rng.uniform(low=-1e-2, high=1e-2, size=x.shape)
+        y = np.arange(20, 10, -1)
+        values = np.outer(x, y)
+
+        def return_arg(target, *args, **kwargs):
+            """A dummy that returns the first argument"""
+            return target
+
+        # Wrong x coordinates should cause an error
+        discharge = create_data_array(x_var_big, y, values, "discharge")
+        loc = create_data_array(x, y, values, "loc")
+        self.assertFalse(discharge.equals(loc))
+        with self.assertRaises(ValueError) as cm:
+            return_period(discharge, loc, loc)
+        self.assertIn(
+            "Reindexing 'loc' to 'discharge' exceeds tolerance!", str(cm.exception)
+        )
+
+        # Small deviations should cause an error if reindexing does not work
+        discharge = create_data_array(x_var, y, values, "discharge")
+        self.assertFalse(discharge.equals(loc))
+
+        # Mock a DataArray
+        da_mock = MagicMock(spec_set=xr.DataArray)
+        da_mock.reindex_like.return_value = loc  # Return without reindexing
+        da_mock.count.return_value = 0  # Mock the count
+
+        # Patch the reindexing
+        with patch(
+            "climada_petals.hazard.rf_glofas.transform_ops.reindex", new=return_arg
+        ):
+            with self.assertRaises(ValueError) as cm:
+                return_period(discharge, da_mock, loc)
+            self.assertIn("cannot align objects", str(cm.exception))
+            self.assertIn("longitude", str(cm.exception))
+
+        # Call the function again, slicing and reindexing should work as expected
+        x_loc = np.arange(11)
+        y_loc = np.arange(25, 5, -1)
+        values_loc = np.outer(x_loc, y_loc)
+        loc = create_data_array(x_loc, y_loc, values_loc, "loc")
+        result = return_period(discharge, loc, loc)
+        self.assertEqual(result.name, "Return Period")
+
+        # "-1" would be a sign that indexing does not work
+        self.assertFalse((result == -1).any())
+
+        # NOTE: This checks if slicing works through 'cdf_mock'
+        # NOTE: Needs 'allclose' because of float32 dtype
+        npt.assert_allclose(result.values, np.ones_like(result.values))
+        npt.assert_allclose(result["longitude"].values, x, atol=1e-8)
+        npt.assert_allclose(result["latitude"].values, y, atol=1e-8)
+
+    def test_return_period_resample(self):
+        """Test 'return_period_resample' operation"""
+        # Make more than 2 dims to check handling of 'core_dims'
+        x = np.arange(10)
+        y = np.arange(20, 10, -1)
+        z = np.linspace(0, 5, 11)
+
+        # NOTE: np.multiply.outer does not flatten input
+        values = np.multiply.outer(np.outer(x, y), z).astype("float")
```

Comment:
Easier to read, in my opinion, but no strong feelings:
```suggestion
        values = (x[:, None, None] * y[None, :, None] * z[None, None, :]).astype("float")
```